packages/reactive_ble_mobile/android/src/main/kotlin/com/signify/hue/flutterreactiveble/ble/ReactiveBleClient.kt

```diff
@@ -75,6 +75,7 @@ open class ReactiveBleClient(private val context: Context) : BleClient {
         return rxBleClient.scanBleDevices(
             ScanSettings.Builder()
                 .setScanMode(scanMode.toScanSettings())
+                .setLegacy(false)
```

Comment:
why you set it to `false` here?

Reply:
Yes, it still works with 4.2 advertisings. setLegacy(false) enables the driver to receive extended advertisings (with more than 31 bytes), without removing support to old versions of BLE. (Google named it badly, "setOnlyLegacy" would be better, but anyhow...)
![image](https://user-images.githubusercontent.com/8076962/181496656-fbb9ff9c-198b-452a-aac7-5c85713ab606.png)

With this change, flutter_reactive_ble on android matches the iOS behaviour, returning both legacy and extended advertisings.

https://developer.android.com/reference/android/bluetooth/le/ScanSettings.Builder

Reply:
Thanks for the contribution!